### DISCUSSION TITLE
How to plot UMAP with custom labels?

### DISCUSSION THREAD

**Discussion by MasonDou** [satijalab/seurat#6964] [Q&A]
I am running a single-cell analysis with Seurat, everything goes smoothly when I try to plot UMAP.
I would like to draw UMAP plot with my custom groups(0 day, 3 day, 7 day and 14 day rather than cluster generated automatic).
I use the code: 
sc.all=CreateSeuratObject(data1,meta.data = metadata)  #it is a matrix with 22166 obs and 56420 variables
group<-c(rep("0 day",6778),rep("3 day",14365),rep("7 day",8284),rep("14 day",26989))
sc.all@meta.data$sample_type<-group
sc.all <- RunPCA(sc.all, features = VariableFeatures(object = sc.all))
sc.all <- FindNeighbors(sc.all, dims = 1:10)
sc.all <- FindClusters(sc.all, resolution = 0.5)
sc.all <- RunUMAP(sc.all, dims = 1:10)
DimPlot(sc.all, reduction = "umap",label =T, group.by = metadata$sample_type)

Error: 
Error in `[.data.frame`(data, , group) : undefined columns selected
In addition: Warning message:
In `[[.Seurat`(object, group.by) :
  Cannot find the following bits of meta data: 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 [... truncated]

Could anyone tell me how to fix it, thanks a lot!!!!!!!!!

**Comment by emanuelavilla** [comment]
Hi! 
Group.by needs the metadata to be specified with quote.

DimPlot(sc.all, reduction = "umap",label =T, group.by = "sample_type")
